classes/sample_set.php

```diff
@@ -66,6 +78,16 @@ public function add_sample(\ExcimerLogEntry $sample) {
             $this->samples[] = $sample;
             $this->counter = 0;
         }
+        // Each time a sample is added, recalculate the maxstackdepth for this set.
+        foreach ($this->samples as $sample) {
+            $trace = $sample->getTrace();
+            if ($trace) {
+                $stackdepth = count($trace);
+                if ($stackdepth > $this->maxstackdepth) {
```

Comment:
there was a comment here which was lost from kevin around using max($a, $b) to make this a little simpler